### PR TITLE
Add derivedClass in default template

### PR DESCRIPTION
**Pull request by qinezh**
@superyyrrzz @vicancy @hellosnow @ansyral @vwxyzh @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs [hunk 1]
```diff
@@ -241,17 +246,24 @@ private static ApiBuildOutput FromModel(ItemViewModel model, Dictionary<string,
...
+                                                                      int level = -1)
         {
             if (extractIndex)
             {
                 return uids?.Select((u, i) => ApiBuildOutputUtility.GetReferenceViewModel(u, references, supportedLanguages, i)).ToList();
             }
+            else if (level != -1)
+            {
+                return uids?.Select(u => ApiBuildOutputUtility.GetReferenceViewModel(u, references, supportedLanguages, level)).ToList();
+            }
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs]: how about move these inside if (extractIndex) and also pass in extractIndex"true" for DrivedClasses?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs [hunk 2]
```diff
@@ -241,15 +246,22 @@ private static ApiBuildOutput FromModel(ItemViewModel model, Dictionary<string,
                 Attributes = model.Attributes,
                 Metadata = metadata.Concat(model.Metadata.Where(p => !metadata.Keys.Contains(p.Key))).ToDictionary(p => p.Key, p => p.Value),
             };
+            output.DerivedClasses = GetReferenceList(model.DerivedClasses, references, model.SupportedLanguages, true, output.Level+1);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs]: >+1 [](start = 131, length = 2)

spaces in between

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs [hunk 3]
```diff
@@ -241,15 +246,22 @@ private static ApiBuildOutput FromModel(ItemViewModel model, Dictionary<string,
...
+            return output;
         }
 
         private static List<ApiReferenceBuildOutput> GetReferenceList(List<string> uids,
                                                                       Dictionary<string, ApiReferenceBuildOutput> references,
                                                                       string[] supportedLanguages,
-                                                                      bool extractIndex = false)
+                                                                      bool extractIndex = false,
+                                                                      int level = -1)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs]: >int level = -1 [](start = 70, length = 14)

this option is specifically for DerivedClasses? Use another method instead?

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs]: this method is to share the same logic, the newly added code however adds new logic to the method

---
In reply to: [91427831](https://github.com/dotnet/docfx/pull/973#discussion_r91427831) [](ancestors = 91427831)

Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs]: I think it make sense to add the new logic in the method. Need extract index?  -> yes -> provide the index or compute index ? -> provide the index...

---
In reply to: [91428041](https://github.com/dotnet/docfx/pull/973#discussion_r91428041) [](ancestors = 91428041,91427831)

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs (hunk 3 above)
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutput.cs]: Seems that codeflow has bug that can't show deep-level comments... Copy the comments from codeflow: 
> I think it make sense to add the new logic in the method. Need extract index?  -> yes -> provide the index or compute index ? -> provide the index...

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**